system/jlib/jsort.cpp

```diff
@@ -41,10 +41,7 @@
 
 static bool sortParallel(unsigned &numcpus)
 {
-    static unsigned numCPUs = 0;
-    if (numCPUs==0) {
-        numCPUs = getAffinityCpus();
-    }
+    unsigned numCPUs = getAffinityCpus();
```

Comment:
getAffinityCpus() already caches its value - but can also be reset.  Without it the parallel quick sort spins forever if the affinity is reduced.

Reply:
Is it being reset from a control request or a query request ?

Reply:
It means that sending a <control::setAffinity/> command to roxie after a taskset will update this setting.  Only really useful for internal testing.